src/init.cpp

```diff
@@ -175,6 +174,17 @@ static fs::path GetPidFile(const ArgsManager& args)
     }
 }
 
+static std::optional<util::SignalInterrupt> g_shutdown;
+
+void InitContext(NodeContext& node)
+{
+    assert(!g_shutdown);
+    g_shutdown.emplace();
```

Comment:
If an exception is thrown during construction here,  the program will core dump. How about adding a try-catch block in the various `MakeNodeInit` that populates the correct `exit_status` and returns a `nullptr`?